### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by tkafalas**

### CODE REVIEW

File: common/common-shim/pom.xml [hunk 1]
```diff
@@ -88,5 +96,37 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <groupId>org.apache.avro</groupId>
+      <artifactId>avro</artifactId>
+      <version>${org.apache.avro.version}</version>
+      <!--<scope>provided</scope>-->
```
Comment (VasilinaTerehova) [inline common/common-shim/pom.xml]: here scope should be provided

File: common/common-shim/pom.xml [hunk 2]
```diff
@@ -88,5 +96,37 @@
...
+      <version>1.41.4</version>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>${jackson.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
```
Comment (VasilinaTerehova) [inline common/common-shim/pom.xml]: we shouldn't hardcode such dependency, at least version properties, and also defining in assembly.xml 
https://github.com/pentaho/pentaho-hadoop-shims/pull/529/files#diff-f88d60091735d4b39ab3f1ec0c80218bR21 like here is enough, we should trust what vendor takes. Why is here hardcode?

File: shims/emr52/impl/pom.xml [hunk 3]
```diff
@@ -314,36 +314,72 @@
       <artifactId>hadoop-mapreduce-client-app</artifactId>
       <version>${org.apache.hadoop.version}</version>
       <scope>compile</scope>
+      <exclusions>
```
Comment (VasilinaTerehova) [inline shims/emr52/impl/pom.xml]: defining of avro dependency on top like you did in common-shim will redefine needed version, 
 <dependency>
 +      <groupId>org.apache.avro</groupId>
 +      <artifactId>avro</artifactId>
 +      <version>${org.apache.avro.version}</version>
 +      
 +    </dependency>

**Comment by SergeyTravin** [comment]
The PR was closed since all work was moved to this PR 
https://github.com/pentaho/pentaho-hadoop-shims/pull/543

**Pull request closed without merging**